### PR TITLE
Enabling tracing

### LINKED ISSUE #329: Raise visibility on background analyzer operation
Can we raise messages (in debug builds only) that go the output window to provide a clue the analyzer is running in the background?

Should (or could) debug builds produce log files on disk that can be opportunistically opened to see what's happening during analysis?

Just to make the background analyzer more useful when using it to author/debug new checks.

### PR DESCRIPTION
**Pull request by eddynaka**
Fixes #329

### CODE REVIEW

File: src/Sarif.Sarifer/AnalyzeProjectCommand.cs [hunk 1]
```diff
@@ -80,52 +80,14 @@ private void MenuCommandCallback(object caller, EventArgs args)
...
-                    List<string> targetFiles = GetFiles(project);
+                    List<string> targetFiles = SariferPackageCommand.GetFiles(project);
 
                     this.backgroundAnalysisService.AnalyzeAsync(project.FullName, targetFiles, this.cancellationTokenSource.Token)
                         .FileAndForget(FileAndForgetEventName.BackgroundAnalysisFailure);
                 }
             }
         }
 
-        private static List<string> GetFiles(Project project)
```
Comment (eddynaka) [inline src/Sarif.Sarifer/AnalyzeProjectCommand.cs:91]: merged this in the SariferPAckageCommand that we are already sharing between the two commands

File: src/Sarif.Sarifer/SariferPackage.cs [hunk 2]
```diff
@@ -73,6 +74,11 @@ await this.GetServiceAsync(typeof(SVsShell)).ConfigureAwait(continueOnCapturedCo
                 this.analyzeProjectCommand = new AnalyzeProjectCommand(mcs);
             }
 
+            if (await GetServiceAsync(typeof(SVsOutputWindow)).ConfigureAwait(continueOnCapturedContext: true) is IVsOutputWindow output)
+            {
+                outputWindowTraceListener = new OutputWindowTraceListener(output, "Sarifer");
```
Comment (eddynaka) [inline src/Sarif.Sarifer/SariferPackage.cs:79]: >outputWindowTraceListener = new OutputWindowTraceListener(output, "Sarifer"); [](start = 16, length = 77)

with this, when we execute and use Trace.Write / Trace.WriteLine, we would see a new dropdown in the output windows with the data that we trace. Here I do not know if we should ship on Release mode. What do you think?

File: src/Sarif.Sarifer/SpamBackgroundAnalyzer.cs [hunk 3]
```diff
@@ -54,6 +56,8 @@ protected override bool AnalyzeCore(Uri uri, string text, string solutionDirecto
                 return false;
             }
 
+            Trace.WriteLine($"Rules loaded: {this.rules.Count}");
```
Comment (eddynaka) [inline src/Sarif.Sarifer/SpamBackgroundAnalyzer.cs:59]: >Trace.WriteLine($"Rules loaded: {this.rules.Count}"); [](start = 12, length = 53)

Adding a trace to know how many rules did we load

File: src/Sarif.Sarifer/SpamBackgroundAnalyzer.cs [hunk 4]
```diff
@@ -69,6 +73,9 @@ protected override bool AnalyzeCore(Uri uri, string text, string solutionDirecto
 
                 // Filtering file before analyzing.
                 IEnumerable<Skimmer<AnalyzeContext>> applicableSkimmers = AnalyzeCommand.DetermineApplicabilityForTargetHelper(context, this.rules, disabledSkimmers);
+
+                Trace.WriteLine($"Rules filtered: {applicableSkimmers.Count()}");
```
Comment (eddynaka) [inline src/Sarif.Sarifer/SpamBackgroundAnalyzer.cs:77]: >Trace.WriteLine($"Rules filtered: {applicableSkimmers.Count()}"); [](start = 16, length = 65)

Adding a trace to know how many rules will be used to analyze the file

File: src/Sarif.Sarifer/OutputWindowTracerListener.cs [hunk 5]
```diff
@@ -0,0 +1,67 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
```
Comment (michaelcfanning) [inline src/Sarif.Sarifer/OutputWindowTracerListener.cs:2]: >Licensed [](start = 3, length = 8)

file name doesn't match class type.

File: src/Sarif.Sarifer/SariferPackage.cs (hunk 2 above)
Comment (michaelcfanning) [reply src/Sarif.Sarifer/SariferPackage.cs:79]: Fine for now! This whole package is experimental.

---
In reply to: [550914135](https://github.com/microsoft/sarif-visualstudio-extension/pull/338#discussion_r550914135) [](ancestors = 550914135)

File: src/Sarif.Sarifer/SpamBackgroundAnalyzer.cs (hunk 3 above)
Comment (michaelcfanning) [reply src/Sarif.Sarifer/SpamBackgroundAnalyzer.cs:59]: Move this to line 50. Replace it with a message "Analyzing {uri}..."

---
In reply to: [550914172](https://github.com/microsoft/sarif-visualstudio-extension/pull/338#discussion_r550914172) [](ancestors = 550914172)

**Review by michaelcfanning — APPROVED** [review]
:shipit:

**Pull request merged**